rfcs/0098-community-team.md

```diff
@@ -0,0 +1,190 @@
+---
+feature: community-team
+start-date: 2021-07-31
+author: Irenes (Irene Knapp)
+co-authors: ashkitten
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Establish an official community team to make moderation decisions, model social norms, and mediate interpersonal problems.
+
+# Motivation
+[motivation]: #motivation
+
+There is not currently any official mechanism for moderation action. It's not sustainable to have to call on Graham any time there's a spammer or conflict that requires moderation, and we'd like to help the community become more self-regulating.
+
+# Goals
+[goals]: #goals
+
+* Model and enforce social norms
+* Mediate disputes when doing so is beneficial to the community
+* Respect the privacy of community members
+* Foster newcomers and develop their leadership potential
+* Moderate all places where community members interact, including Discourse, Matrix, and GitHub
+* Ensure that collaboration on shared technical goals is possible
+* Ensure that the Nix community is a welcoming space for people from marginalized backgrounds
+* Do not allow the Nix community to be a place for spreading ideas rooted in fascism or bigotry
+* Ensure that people in positions of authority within the community, whether technical or otherwise, promote community participation in all things and use their authority in ways that are consistent with the community's values
+* Don't be overly driven by the personalities of a small set of people
```

Comment:
Note that this thing you are trying to avoid is exactly the real source of the perception of moderation as done reasonably (because it is done by people already trusted for reasonas unrelated to moderation), so maybe there needs to be some mention of an alternative source of legitimacy among goals?

Reply:
> Note that this thing you are trying to avoid is exactly the real source of the perception of moderation as done reasonably (because it is done by people already trusted for reasonas unrelated to moderation), so maybe there needs to be some mention of an alternative source of legitimacy among goals?

To be clear: You're saying that you believe being driven by the personalities of a small set of people is the source of trust?

I don't agree with that.

I think people's personalities are relevant to trust, and I guess we agree on that part. I think that each member of the community team needs to put in the work to build trust with the community. I think that some of that work should happen before they become part of the team, and how successful they've been should be evaluated as part of adding them to the team. I think that some of that work should happen while they're on the team, and team members should help each other to do it. I also think that building trust is something everyone, in all aspects of their life, should always be trying to do - but I didn't say that part in the document because it's out of scope.

I don't think that trust has to be about it always being the *same* people. I think we can teach these skills, and we should. If it were always driven by the same people (for example, me and ashkitten), there would always be questions about how well we're balancing the values as set forth in this document against our personal beliefs and flaws. We can't eliminate that concern but I hope that we can at least describe a process which minimizes the opportunity for it, and which shows good faith from how it's structured. The goal is to create something that's bigger than any one person.

Reply:
> I think that each member of the community team needs to put in the work to build trust with the community

I know a lower bound on the amount of committed effort of Graham or Ryan. And almost everyone knows. And this surely creates trust — and does not really scale to a rotating team.

Reply:
I agree. The rotating team scales in an entirely different way. It's still based on trust, but the trust isn't coming from quite the same place, although there's overlap. That's by design. What Graham can't do is be everywhere, all the time. The team can't be *everywhere*, but it can be in more places, and there are other reasons why we think it's both more scalable and more sustainable. Let me try to explain. :)

Because we encourage everyone in the community to think of themselves as potential leaders, and set up community team membership as something to aspire to by making it clear that the community team members are supposed to set the best example they can, we'll always have a talent pool available - we're creating the talent pool as we go.

Because former community team members will still be part of the overall community, we're ensuring that the skills to defuse tense situations will be spread throughout the community. Thus we hope that over time, there will be fewer escalations, and less need for the type of leadership that only long-tenured charismatic figureheads can provide.

(I hope Graham and Ryan don't mind me calling them "long-tenured charismatic figureheads". I've been a bit of one myself, in other places, and I don't mean to say that it's a bad thing. It's only that it's not sustainable. Nobody can keep going forever, we have to always be training our replacements.)

Reply:
> Because we encourage everyone in the community to think of themselves as potential leaders,

This statement in its own — as it is made — is already repulsive to me. I am here to solve technical issues and help people avoid predictable coordination problems, not to be a leader of whatever.

> The team can't be everywhere, but it can be in more places

There are of course more widely respected members with a long track record of effort into developing who do moderation and community positioning work.

> It's still based on trust, but the trust isn't coming from quite the same place, although there's overlap.

Trust in long term commitment cannot be established quickly. Wide trust cannot be established quickly.

Reply:
> > Because we encourage everyone in the community to think of themselves as potential leaders,
> 
> This statement in its own — as it is made — is already repulsive to me. I am here to solve technical issues and help people avoid predictable coordination problems, not to be a leader of whatever.

I'm happy that we've identified something concrete that we disagree on. It's of course totally fine to disagree. I think it will be helpful for anyone who's reading along trying to form an opinion to know that, yes, the authors of the RFC specifically do want to encourage newcomers and treat everyone as potential leaders, and that that is something that not everyone is on the same page about. I think everyone who's gotten this far in the thread should take a moment to decide for themselves how they feel about whether the Nix community should be explicitly trying to do leadership development.

Please know that of course neither you, nor anyone else, would be forced into a leadership role you didn't want. It is my belief that any software project is also a community. That's unavoidable, whether it's intended or not. I think it makes sense for people who care about community building and to explicitly think about software as a community undertaking, not just as bits and bytes.

> Trust in long term commitment cannot be established quickly. Wide trust cannot be established quickly.

Yep! Very much agreed. The first paragraph under "how new members join and leave" speaks to this, in that it implies that people joining the community team should already have some measure of trust, although it's attended to be an attainable bar for people who don't already have some sort of seniority.

> New members may apply to join the community team, and existing team members will decide on these applications. The team will evaluate applicants by the length of time they've been part of the Nix community, by their good character, by their demonstrated ability to exercise good judgement, and by their commitment to public service and a willingness to be self-critical. To form the initial team, the founding members will be appointed by Graham at his sole discretion.

https://github.com/IreneKnapp/rfcs/blob/community/rfcs/0098-community-team.md#how-new-members-join-and-leave-the-community-team

Reply:
> Please know that of course neither you, nor anyone else, would be forced into a leadership role you didn't want. 

There is a spectrum. Maybe something along the lines: might be invited, always welcome, always invited, expected, forced. I am not sure it is a good idea for a team tasked with moderation to be always-invited; but I also feel this proposal as written tries to push from there to «expected». I confidently do not find this good.

> It is my belief that any software project is also a community. 

Your proposal is pushing the community ti be a tighter-knit one. I believe this project benefits from as loose a community as feasible while enforcing some set of basic norms of respect.

Reply:
> There is a spectrum. Maybe something along the lines: might be invited, always welcome, always invited, expected, forced. I am not sure it is a good idea for a team tasked with moderation to be always-invited; but I also feel this proposal as written tries to push from there to «expected». I confidently do not find this good.

The intent is that, as a community, we will try to see the potential in everybody, and the community team will have the responsibility of encouraging that potential. Nobody will be pushed towards leadership that they aren't ready for or have expressed that they don't want; that would be at odds with the value of inclusion.

I think your concern is a pretty fair one but I don't think that our actual goal is anywhere on that linear spectrum you're offering, so I don't know quite how to talk about it in reference to that spectrum.

> Your proposal is pushing the community ti be a tighter-knit one. I believe this project benefits from as loose a community as feasible while enforcing some set of basic norms of respect.

Yes, it is pushing the community to be somewhat more tightly-knit than it is presently. I do see that as desirable. At the same time, of course, it will remain a globally distributed loose affiliation of anyone who wants to spend time on it, and all the Nix source repositories will continue to be fully open to the general public. what counts as tight-knit for an open source project is still quite loose in the big picture.

I'm sure you do realize that, and I'm not trying to trivialize your concern, only to explain why we think the trade-off is worth it.

Reply:
> I think your concern is a pretty fair one but I don't think that our actual goal is anywhere on that linear spectrum you're offering, so I don't know quite how to talk about it in reference to that spectrum.

I think it is «encouraged to» (done reasonably) on the spectrum I described, and I think it is going slightly too far although it has a chance of avoiding doing damage.

> I'm sure you do realize that, and I'm not trying to trivialize your concern, only to explain why we think the trade-off is worth it.

I do not see an explanation _why_ somewhat tighter-knit is an improvement specifically for maintaining Nixpkgs in your message.

Reply:
I think these posters make a very valid point.
> I'd like to respond to this:
> 
> > I'd like to comment on a fascism. As someone that lives in a country that was occupied by nazis and fascist (and my grandfather being shot by one), I'll allow myself to have an opinion. The combination of lack of clear rules and power to remove people from the community is exactly the kind of environment such ideas can prosper in. It doesn't mean it will happen, but it has the conditions to deal with political opponents. Writing down guidelines and limits creates a safe space for everyone and makes sure we can have an honest discussion about the policy.
> 
> It was the original authors' statement about Fascism that brought about my engagement with this RFC.
> 
> Fascism is a subset of socialism (read the [platforms and communications](https://en.wikipedia.org/wiki/Fascist_Manifesto) if you don't believe me). All Socialist dictators that I can think of are easily labeled fascists. Similarly they are all labeled, usually by themselves, Communists as well.
> 
> Communism did not merely kill. Communists also persecuted, starved, enslaved, exiled, and imprisoned vast numbers of their populations, based on religion, ethnicity, sexuality, and other forms of identity. I know these things not only because I have a BS in Political Science, and read Aleksandr Solzhenitsyn extensively, but my wife grew up in the USSR and experienced it first hand.
> 
> I must insist that any code of conduct references to fascists must include references to communists as well.
> 
> I am very much anti-fascist in the generic sense of the term, but it seems to me that _most_ anti-fascists and those who trumpet their position on the matter do so to distract the population from the vastly greater quantity of evils done by communists - because they _are_ communists.
> 
> I am a committed member of this community. I care very deeply for its reputation. I am also sensitive to the perception of the community by the outside world. I will continue to insist that any statements on anti-fascism must include statements on anti-communism as well, lest we be thought to be communists ourselves.
> 
> Otherwise, I suggest dropping political terms like "fascism" and "communism" altogether.
> 
> I suggest we vote. Thumbs down if you think we should drop terms like "fascism" and "communism". Thumbs up if you think we should include them. Respond in defense of communism if you think we should mention "fascism" and not "communism."

I think reactions like this are an example of why making specific statements about ideology is dangerous, because they are actually not as specific and cut in stone as one may think. And further they are usually local. Nix and NixOS however does well to draw from a global pool of developers, so as this reply makes clear, perhaps it would be better to not make any statements that can be seen as taking a political stand, even if they in some contexts might seem *obviously right*, they may not do so in all cultures and societies. The core message would be better to simply clarify the **rationale** behind prohibiting *fascist ideology* and instead focus on the latter message, namely to sanction *bigotry*, not just because it's "right", but because it's a major roadblock to a thriving community of diverse developers.

And likewise of course, to make a more defined framework for what constitutes bigotry, as it can, as another poster points out, easily become a rubber stamp for questionable decisions.

When we look away from the specific words being used, it's clear that we are all interested in creating a community that is open and inclusive, and with a healthy application of good faith, I'm sure the mention of such a inflammatory ideologies could be tactfully avoided.

If anything I would actually prefer if there was a statement to the effect that Nix and NixOS should not be political projects, and repeated infractions would lead to some sort of sanctions, of course, when it was not done in the process of deliberating an actual issue facing the project, such as e.g. FOSS aspects of packaging, or the RFC process, for instance.